src/sonarqube-gui-plugin/src/main/resources/static/js/cloneViewer/CloneInstanceSelector.js

```diff
@@ -0,0 +1,143 @@
+
+  this.elem = null; // :DOMElement
+  this.id; // : int (sorszám)
+  this.cloneInstanceList; // :CloneClass[]
+  this.selected; // :int
+  this.codeBrowser; // :CodeBrowser (manages the diffs)
+  this.parent; // :CloneClassSelector
+
+  var self = this;
+
+  this.init = function(HTMLelem,options) {
+
+    this.id = (typeof options.id !== "undefined") ? options.id : 0;
+    this.codeBrowser = (typeof options.codeBrowser !== "undefined") ? options.codeBrowser : null;
+    this.parent = (typeof options.parent !== "undefined") ? options.parent : null;
+    this.cloneInstanceList = (typeof options.cloneInstanceList !== "undefined") ? options.cloneInstanceList : [];
+    this.selected = (typeof options.selected !== "undefined") ? options.selected : 0;
+
+    this.elem = HTMLelem;
+    this.bindElement(HTMLelem);
+    this.select(this.selected);
+
+    SM.MetricLoader.subscribe("finishedAllRequests", function() {
+      self.renderMetrics();
+    });
+    this.renderAll();
+  };
+
+  /**
+   * Generates HTML of dependant of the current state of the object.
+   * Then overwrites the HTML in the visible webpage as well.
+   * @return {void}
+   */
+  this.renderAll = function() {
+    this.elem.html("");
+    this.elem.addClass("sm-cloneInstanceSelector");
+    // put together the selectors html string
+    var html = [];
+
+    html.push('<div id="cloneInstanceLinkContainer">');
+      html.push(SM.cloneViewer.generatePositionAnchorPopup(this.cloneInstanceList[this.selected]));
+    html.push('</div>');
+    html.push('<div>');
+      html.push('<select id="selectmenu">');
+      this.cloneInstanceList.forEach(function(cloneInstance, i) {
+          html.push("<option value=" + i + "> " + cloneInstance.name + "</option>");
+      });
+      html.push('</select>');
+    html.push('</div>');
+    html.push('<div id="cloneInstanceMetricsContainer'+this.id+'"></div>');
+
+    this.elem.append(html.join(""));
+
+    this.renderMetrics();
+
+    // make selectmenu
+    $("#cloneInstanceSelector" + this.id + " #selectmenu").selectmenu();
+    $("#cloneInstanceSelector" + this.id + " #selectmenu").val(this.selected).selectmenu("refresh");
+  };
+
+  this.renderMetrics = function() {
+    var div = $('#cloneInstanceMetricsContainer'+this.id);
+    var html = [];
+    var data = this.cloneInstanceList[this.selected].cloneInstanceMetrics;
+    var metrics = SM.state[SM.options.component.key].clone.instanceMetrics;
+    metrics.forEach(function(metric) {
+      html.push(SM.cloneViewer.getFormatedMetric(data[metric.title], metric));
+    });
+    div.html(html.join(""));
+    $("#cloneInstanceSelector" + this.id + " .sm-cloneviewer-metric-title-container").tooltip({
+      content: function() {
+        return $(this).prop('title');
+      }
+    });
+  };
+
+  /**
+   * selects a CloneClass, and calls everything that needs to update its state.
+   * @param  {int} choice    the id of the cloneclass in the this.CloneClassList
+   * @return {void}
```

Comment:
Please add a new line before `@return` and write `undefined` instead of void like before.